### PR TITLE
Environments SPA fixes:

### PR DESCRIPTION
**Pull request by vrushaliwaykole**
Issue: #7176 

Description:

Updated environment link on the dashboard
Added warning message for environments which are not associated with any pipeline or agent
Update of pipeline association was not reflecting properly on UI. Fixed it.

### CODE REVIEW

File: server/src/main/webapp/WEB-INF/rails/webpack/views/pages/new-environments/index.scss [hunk 1]
```diff
@@ -13,28 +13,27 @@
...
 .env-name {
-  font-weight:   600;
-  min-width:     250px;
-  max-width:     250px;
+  font-weight: 600;
+  min-width: 250px;
+  max-width: 250px;
   text-overflow: ellipsis;
-  overflow:      hidden;
+  overflow: hidden;
```
Comment (kritika-singh3) [inline server/src/main/webapp/WEB-INF/rails/webpack/views/pages/new-environments/index.scss]: Please set `align on values` for SCSS file types in our Intellij.

**Comment by vrushaliwaykole** [comment]
Warning message when neither pipelines and agents are associated with the environment - 

Screenshot 2019-11-26 at 2 03 25 PM

File: server/src/main/webapp/WEB-INF/rails/webpack/views/pages/new-environments/environments_widget.tsx [hunk 2]
```diff
@@ -54,27 +55,45 @@ export class EnvironmentWidget extends MithrilViewComponent<EnvAttrs> {
...
+        <div data-test-id="warning-tooltip-content" className={styles.warningTooltipContent}>
+          <p>Neither pipelines nor agents are associated with this environment.</p>
+        </div>
+      </div>;
+    }
+    return [
+      warningButton,
+      <IconGroup>
+        <Delete
+          title={environment.canAdminister() ? undefined : `You dont have permissions to delete '${environment.name()}' environment.`}
```
Comment (kritika-singh3) [inline server/src/main/webapp/WEB-INF/rails/webpack/views/pages/new-environments/environments_widget.tsx]: The message can be: `You are not authorized to delete the '${environment.name()}' environment.` WDYT?

File: server/src/main/webapp/WEB-INF/rails/webpack/views/pages/new-environments/edit_pipelines_modal.tsx [hunk 3]
```diff
@@ -91,11 +92,11 @@ export class UnavailablePipelinesBecauseOfOtherEnvironmentWidget extends Mithril
...
           pipelines.map((pipeline) => {
-            const environmentLink = <span data-test-id={`pipeline-list-item-for-${pipeline.name()}`}
-                                          class={styles.link}>
+            const environmentWithOrigin = vnode.attrs.pipelinesVM.environments.findEnvironmentForPipeline(pipeline.name());
+            const environmentLink       = <span data-test-id={`pipeline-list-item-for-${pipeline.name()}`} class={styles.link}>
               (ENVIRONMENT:
-              <a href="#">
-                {vnode.attrs.pipelinesVM.environments.findEnvironmentForPipeline(pipeline.name())!.name()}
+              <a target="_blank" href={SparkRoutes.editEnvironmentsPath(environmentWithOrigin!.name())}>
+                {environmentWithOrigin!.name()}
```
Comment (kritika-singh3) [inline server/src/main/webapp/WEB-INF/rails/webpack/views/pages/new-environments/edit_pipelines_modal.tsx]: A suggestion (can be ignored): Can we use the `Link` component here?

File: server/src/main/webapp/WEB-INF/rails/webpack/helpers/spark_routes.ts [hunk 4]
```diff
@@ -439,4 +439,8 @@ export class SparkRoutes {
   static editTemplatePermissions(templateName: string) {
     return `/go/admin/templates/${templateName}/permissions`;
   }
+
+  static editEnvironmentsPath(environmentName: string) {
```
Comment (GaneshSPatil) [inline server/src/main/webapp/WEB-INF/rails/webpack/helpers/spark_routes.ts]: is this an `editEnvironmentsPath` ? or a `getEnvironmentPathOnSPA`?

Comment (vrushaliwaykole) [reply server/src/main/webapp/WEB-INF/rails/webpack/helpers/spark_routes.ts]: Changed as the part of [this](https://github.com/gocd/gocd/pull/7320/commits/aaac131674ca254d41d8957a7545bb3beb5d7c01) commit.

**Pull request merged**